src/NixTree/StorePath.hs

```diff
@@ -68,11 +71,14 @@ data StoreName s = StoreName NixStore Text
 
 mkStoreName :: NixStore -> FilePath -> Maybe (StoreName a)
 mkStoreName ns path = do
-  let ps = unNixStore ns
+  let ps' = unNixStore ns
+      -- edge case: for ps' == "https://cache.nixos.org",
+      -- the prefix is still "/nix/store"
+      ps = if ps' `isPrefixOf` path then ps' else "/nix/store/"
```

Comment:
Thanks for the comment, it was helpful!

But it kind of suggests me that we are overloading `--store`? It feels like we're both using it for `storeDir`, and also for the remote store. But `https://cache.nixos.org` store uses `/nix/store` as a remote store so we're working around it. But I can imagine there can be other remote stores that use another store path.

What do you think?

Reply:
I guess I am also wondering why do we need to do this on top of passing `--store` to `nix path-info` above?

Reply:
This is no longer necessary, see the updated code!